src/storage/src/source/mod.rs

```diff
@@ -408,9 +411,55 @@ pub(crate) trait SourceReader {
     ///
     /// Note that implementers are required to present messages in strictly ascending\
     /// offset order within each partition.
+    async fn next(
+        &mut self,
+    ) -> Option<Result<SourceMessage<Self::Key, Self::Value>, SourceReaderError>> {
+        // Compatiblity implementation that delegates to the deprecated [Self::get_next_method]
+        // call. Once all source implementations have been transitioned to implement
+        // [SourceReader::next] directly this provided implementation should be removed and the
+        // method should become a required method.
+        loop {
+            match self.get_next_message() {
+                Ok(NextMessage::Ready(msg)) => return Some(Ok(msg)),
+                Err(err) => return Some(Err(err)),
+                // There was a temporary hiccup in getting messages, check again asap.
+                Ok(NextMessage::TransientDelay) => {
+                    tokio::time::sleep(Duration::from_millis(1)).await
```

Comment:
why wait at all? we immediately re-activate the operator in this case currently

Reply:
We need to yield control back to timely to run other operators so we must produce a `Poll::Pending` state for the timely closure to exit. I wasn't sure if `Duration::from_millis(0)` was special cased, that's why I put `1` here. It should go away once we migrate things to use `next()`